### DISCUSSION TITLE
The speaker functionality is pretty good, but there are some problems which I think could be solved with a few additions

### DISCUSSION THREAD

**Discussion by rilhia** [deepgram/community#104] [Feature Requests]
In the first scenario I have tested (a group meeting with a single recording of 1 channel or 2 for stereo) sometimes a single speaker who is not being interrupted can be split into 2 different speakers. So something like ....

"Hello, good morning, afternoon or evening!" 

....can be split into ....

Speaker 0: Hello
Speaker 1: Good
Speaker 1: Morning
Speaker 1: Afternoon
Speaker 1: Or
Speaker 1: Evening

When you go to listen to the words and compare "Hello" against the voice of Speaker 0, it is clearly not that speaker. Now this doesn't happen too much and usually appears to happen at the beginning of sentences when it does. I have tried to use the speaker compliance value to work with this, but it only says how likely the speaker is correct, if I understand this correctly.

The second scenario is more of an issue and I imagine much harder to solve....especially if the first issue remains. If you have a series of podcasts to work with (this is what I am using) and predominantly all the same presenters (3 or 4), then if they do not start each podcast speaking in the same order, it is very difficult to work with the data without listening to the podcasts and then working out who speaker 0,1,2 and 3 are. Then, building that into your app. It takes a lot of manual effort to align the people to the speaker values.

Now, I know that this could be mitigated for by ensuring that each presenter introduces themselves in order at the beginning of each podcast. This *should* work, but as I pointed out in the first issue, this is not guaranteed to work....especially at the beginning of a sentence. I suspect (I may be wrong) that this occurs due to the fact that no other speaker has been heard at this point and any slight variance in someone's voice could trigger a "new speaker" recognition incorrectly.

What I am going to ask is, is it possible to give more info on the speakers? So information on speech speed, tone, inflection, etc? People tend to speak in different ways to each other, but with a consistency in how THEY speak. This is what banking voice recognition authentication is based on. If something like this could be provided, it may allow users to solve these problems themselves.

This is certainly not a complaint and I am really quite impressed with what I have been able to achieve with this so far. But with this extra data, I think that this software could be used for many more use cases. Maybe even security, as I mentioned above.

**Comment by lukeocodes** [comment]
Closing as a duplicate of https://github.com/orgs/deepgram/discussions/176
